src/main/java/org/opendaylight/yangtools/triemap/ImmutableKeySet.java

```diff
@@ -37,7 +36,10 @@
 
     @Override
     public Iterator<K> iterator() {
-        return Iterators.transform(map().immutableIterator(), Entry::getKey);
+        ImmutableIterator<K, ?> iterator = map().immutableIterator();
+        ArrayList<K> keyList = new ArrayList();
+        iterator.forEachRemaining(a -> keyList.add(a.getKey()));
```

Comment:
Again eager instantiation -- what if happens when this map has a million entries and user wants just the third one (for whatever reason)?